ResearchKit/Common/ORKFormItemCell.m

```diff
@@ -1148,3 +1156,168 @@ - (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRang
 }
 
 @end
+
+
+#pragma mark - ORKFormItemLocationCell
+
+@interface ORKFormItemLocationCell () <ORKLocationSelectionViewDelegate>
+
+@property (nonatomic, assign) BOOL editingHighlight;
+@property (nonatomic, strong) NSString *placeholder;
+
+@end
+
+
+@implementation ORKFormItemLocationCell {
+    ORKLocationSelectionView *_selectionView;
+    NSLayoutConstraint *_heightConstraint;
+}
+
+- (void)cellInit {
+    [super cellInit];
+    
+    _selectionView = [[ORKLocationSelectionView alloc] initWithOpenMap:NO
+                                                    useCurrentLocation:((ORKLocationAnswerFormat *)self.formItem.answerFormat).useCurrentLocation
+                                                edgeToEdgePresentation:NO];
+    _selectionView.delegate = self;
+    
+    if (_placeholder != nil) {
+        [_selectionView setPlaceholderText:_placeholder];
+    }
+    
+    [self applyAnswerFormat];
+    [self answerDidChange];
+    
+    [self.contentView addSubview:_selectionView];
+    
+    [self setUpConstraints];
+}
+
+- (void)setUpConstraints {
+    NSMutableArray *constraints = [NSMutableArray new];
+    
+    NSDictionary *dictionary = @{@"_selectionView":_selectionView};
+    ORKEnableAutoLayoutForViews([dictionary allValues]);
+    NSDictionary *metrics = @{@"verticalMargin":@(10), @"horizontalMargin":@(self.separatorInset.left), @"verticalMarginBottom":@(10.0 - (1.0 / [UIScreen mainScreen].scale))};
```

Comment:
@YuanZhu-apple The vertical margin bottom is set to be 10 points minus the height of a cell separator. This is what was causing the layout issues before. I believe this could be considered a bug with how table view cell heights are calculated, or just not documented. I do believe there should be a better way "retrieve" the height of the cell separator, as its height may change in other iOS versions.